middlewares/metrics.go

```diff
@@ -22,6 +23,52 @@ type RetryMetrics interface {
 	getRetryCounter() metrics.Counter
 }
 
+// MultiMetrics is a struct that provides a wrapper container for multiple Metrics, if they are configured
+type MultiMetrics struct {
+	wrappedMetrics       *[]Metrics
+	reqsCounter          metrics.Counter
+	reqDurationHistogram metrics.Histogram
+	retryCounter         metrics.Counter
+}
+
+// NewMultiMetrics creates a new instance of MultiMetrics
+func NewMultiMetrics(manyMetrics []Metrics) *MultiMetrics {
+	counters := []metrics.Counter{}
+	histograms := []metrics.Histogram{}
+	retryCounters := []metrics.Counter{}
+
+	for _, m := range manyMetrics {
+		counters = append(counters, m.getReqsCounter())
+		histograms = append(histograms, m.getReqDurationHistogram())
+		retryCounters = append(retryCounters, m.getRetryCounter())
+	}
+
+	var mm MultiMetrics
+
+	mm.wrappedMetrics = &manyMetrics
+	mm.reqsCounter = multi.NewCounter(counters...)
+	mm.reqDurationHistogram = multi.NewHistogram(histograms...)
+	mm.retryCounter = multi.NewCounter(retryCounters...)
+
+	return &mm
+}
+
+func (mm *MultiMetrics) getReqsCounter() metrics.Counter {
+	return mm.reqsCounter
+}
+
+func (mm *MultiMetrics) getReqDurationHistogram() metrics.Histogram {
+	return mm.reqDurationHistogram
+}
+
+func (mm *MultiMetrics) getRetryCounter() metrics.Counter {
+	return mm.retryCounter
+}
+
+func (mm *MultiMetrics) getWrappedMetrics() *[]Metrics {
```

Comment:
Yep, you are correct... Same PR that added retryMetrics, removed the `getWrappedMetrics()` method.  It was originally present on the initial `Metrics` interface.  Will clean-up